lib/trino-parquet/src/test/java/io/trino/parquet/TestTupleDomainParquetPredicate.java

```diff
@@ -205,26 +207,71 @@ public void testShortDecimal()
+    }
+
+    @Test
+    public void testShortDecimalWithNoScale()
+            throws Exception
+    {
+        String column = "ShortDecimalColumnWithNoScale";
         Type type = createDecimalType(5, 0);
         assertEquals(getDomain(type, 0, null, ID, column, UTC), all(type));
 
         assertEquals(getDomain(type, 10, longColumnStats(100L, 100L), ID, column, UTC), singleValue(type, 100L));
+        // Test that statistics overflowing the size of the type are not used
+        assertEquals(getDomain(type, 10, longColumnStats(123456L, 123456), ID, column, UTC), notNull(type));
 
         assertEquals(getDomain(type, 10, longColumnStats(0L, 100L), ID, column, UTC), create(ValueSet.ofRanges(range(type, 0L, true, 100L, true)), false));
         // fail on corrupted statistics
         assertThatExceptionOfType(ParquetCorruptionException.class)
                 .isThrownBy(() -> getDomain(type, 10, longColumnStats(100L, 10L), ID, column, UTC))
-                .withMessage("Corrupted statistics for column \"ShortDecimalColumn\" in Parquet file \"testFile\": [min: 100, max: 10, num_nulls: 0]");
+                .withMessage("Corrupted statistics for column \"ShortDecimalColumnWithNoScale\" in Parquet file \"testFile\": [min: 100, max: 10, num_nulls: 0]");
     }
 
     @Test
     public void testLongDecimal()
             throws Exception
     {
         String column = "LongDecimalColumn";
+        DecimalType type = createDecimalType(20, 5);
+        BigInteger maximum = new BigInteger("12345678901234512345");
+
+        Slice zero = unscaledDecimal(BigInteger.valueOf(0L));
+        Slice hundred = unscaledDecimal(BigInteger.valueOf(100L));
+        Slice max = unscaledDecimal(maximum);
+
+        assertEquals(getDomain(type, 0, null, ID, column, UTC), all(type));
+        assertEquals(getDomain(type, 10, binaryColumnStats(maximum, maximum), ID, column, UTC), singleValue(type, max));
+
+        assertEquals(
+                getDomain(type, 10, binaryColumnStats(0L, 100L), ID, column, UTC),
+                create(ValueSet.ofRanges(range(type, zero, true, hundred, true)), false));
+        // fail on corrupted statistics
+        assertThatExceptionOfType(ParquetCorruptionException.class)
+                .isThrownBy(() -> getDomain(type, 10, binaryColumnStats(100L, 10L), ID, column, UTC))
+                .withMessage("Corrupted statistics for column \"LongDecimalColumn\" in Parquet file \"testFile\": [min: 0x64, max: 0x0A, num_nulls: 0]");
+    }
+
+    @Test
+    public void testLongDecimalWithNoScale()
+            throws Exception
+    {
+        String column = "LongDecimalColumnWithNoScale";
         DecimalType type = createDecimalType(20, 0);
-        Slice zero = encodeScaledValue(new BigDecimal("0"), type.getScale());
-        Slice hundred = encodeScaledValue(new BigDecimal("100"), type.getScale());
+        Slice zero = unscaledDecimal(BigInteger.valueOf(0L));
+        Slice hundred = unscaledDecimal(BigInteger.valueOf(100L));
```

Comment:
The expectations here didn't match the code. The encoding is done here using `unscaledDecimal` https://github.com/trinodb/trino/blob/master/lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java#L264